pkg/transformers/json/transformer.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package json
+
+import (
+	"encoding/json"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/transformers"
+)
+
+const (
+	// ContentType represents JSON format content type.
+	ContentType = "application/json"
+)
+
+var (
+	errDecode = errors.New("failed to decode json")
+)
+
+type transformer struct {
+	keys []string
+}
+
+// New returns transformer service implementation for SenML messages.
+func New(keys []string) transformers.Transformer {
+	return transformer{
+		keys: keys,
+	}
+}
+
+func (t transformer) Transform(mm messaging.Message) (interface{}, error) {
+	var payload map[string]interface{}
+	err := json.Unmarshal(mm.Payload, &payload)
+	if err != nil {
+		return []transformers.Message{}, errors.Wrap(errDecode, err)
+	}
+
+	var msgs []transformers.Message
+
+	for k, v := range payload {
+		// Apply key filter
+		for _, tk := range t.keys {
+			if tk == "*" || tk == k {
+				kvmgs := createKeyValMsg(mm, k, v)
+				msgs = append(msgs, kvmgs...)
+				continue
+			}
+		}
+	}
+
+	return msgs, nil
+}
+
+func createKeyValMsg(mm messaging.Message, key string, val interface{}) []transformers.Message {
+	msg := transformers.Message{
+		Channel:   mm.Channel,
+		Subtopic:  mm.Subtopic,
+		Publisher: mm.Publisher,
+		Protocol:  mm.Protocol,
+		Name:      key,
```

Comment:
is there a way to group records by incoming message, as I understand incoming JSON payload that has like 10 fields will produce 10 records in writers database, but do we have a way to group these 10 records?

Reply:
what I meant is that you can easily determine that group of 10 records is from one `JSON` message.

Reply:
Yes, for example with a reader JSON transformer that will group messages by timestamp (kind of state) and re-create the corresponding JSON